app/src/main/java/com/example/concordia_campus_guide/Activities/RoutesActivityViewModel.java

```diff
@@ -39,4 +49,44 @@ public void setFrom(Place from) {
         this.from = from;
     }
 
+    public void setShuttles() {
+        shuttles = appDB.shuttleDao().getAll();
+    }
+
+    public List<Shuttle> getShuttles() {
+        String campusFrom = "";
+        String campusTo = "";
+        if (getFrom() != null && getTo() != null) {
+            if (from.getClass() == Building.class) {
+                campusFrom = ((Building) from).getCampus();
+            } else if (to.getClass() == Building.class) {
+                campusTo = ((Building) to).getCampus();
+            } else if (from.getClass() == Floor.class) {
+                campusFrom = ((Floor) from).getCampus();
+            } else if(to.getClass() == Floor.class) {
+                campusTo = ((Floor) to).getCampus();
+            }
+            if (campusFrom.compareTo(campusTo) == 0) {
+                return null;
+            }
+        }
+
+        Calendar calendar = Calendar.getInstance();
+        String day = new SimpleDateFormat("EEEE", Locale.ENGLISH).format(calendar.getTime());
+        SimpleDateFormat time = new SimpleDateFormat("HH:mm");
+        shuttles = appDB.shuttleDao().getScheduleByCampusAndDayAndTime(campusFrom, day, time.format(calendar.getTime()));
+        return shuttles;
+    }
+
+    public String getShuttleDisplayText(List<Shuttle> shuttles) {
+        String content = "";
+        if (shuttles == null || shuttles.size() == 0) {
+            return "No available routes using the shuttle service";
+        }
+        String campusTo = shuttles.get(0).getCampus().compareTo("SGW") == 0 ? "LOY" : "SGW";
```

Comment:
is this necessary? isn't getCampus() already returning "LOY" or "SGW"?

Reply:
Yes but getCampus() is for the origin campus, this line is returning the destination campus. It's just for the UI

Reply:
oh so it inverts it
I get it cool